themes/qgrid/source/js/menu.js

```diff
@@ -15,26 +15,196 @@ function toggleVisibility(e) {
+	for (let i = tags.length - 1; i >= 0; i--) {
+		if (tags[i]) {
+			const tag = tags[i].split(',');
+			let dataFound;
+			next: for (let j = 0; j < tag.length; j++) {
+				const foundTag = tag[j].split(' ');
+				for (let k = 0; k < foundTag.length; k++) {
+					const index = foundTag[k].toLowerCase().indexOf(search);
+					const title = li[i].getElementsByClassName('title')[0];
+					if (index > -1 && li[i] != active && search != '') {
+						li[i].style.display = '';
+						title.style.width = '50%';
+						title.style.display = 'inline';
+						title.style.font = 'italic normal 10pt arial'
+						title.style.paddingLeft = '8.8%';
+						title.style.color = "grey";
+						if (li[i].getElementsByTagName('data-found').length === 0) {
+							title.prepend("/ ");
+							let select = document.createElement('data-found');
+							let link = document.createElement('a');
+							if (foundTag.length > 2 && k != foundTag.length - 1 && k != foundTag.length - 2) {
+								link.innerHTML = foundTag[k] + ' ' + foundTag[k + 1] + ' ...';
+							}
+							else if (foundTag.length > 2 && (k == foundTag.length - 1 || k == foundTag.length - 2)) {
+								link.innerHTML = '... ' + foundTag[k - 1] + ' ' + foundTag[k];
+							}
+							else {
+								link.innerHTML = tag[j];
+							}
+							select.appendChild(link);
+							li[i].insertBefore(select, li[i].children[0]);
+							dataFound = li[i].getElementsByTagName('data-found')[0];
+							const a = dataFound.getElementsByTagName('a')[0];
+							dataFound.style.display = 'inline';
+							a.style.display = 'inline';
+							a.setAttribute('href', title.getAttribute('href'));
+						}
+						else {
+							if (title.innerHTML.indexOf("/ ") == -1) {
+								title.prepend("/ ");
+							}
+							dataFound = li[i].getElementsByTagName('data-found')[0].getElementsByTagName('a')[0];
+							if (foundTag.length > 2 && k != foundTag.length - 1 && k != foundTag.length - 2) {
+								dataFound.innerHTML = foundTag[k] + ' ' + foundTag[k + 1] + ' ...';
+							}
+							else if (foundTag.length > 2 && (k == foundTag.length - 1 || k == foundTag.length - 2)) {
+								dataFound.innerHTML = '... ' + foundTag[k - 1] + ' ' + foundTag[k];
+							}
+							else {
+								dataFound.innerHTML = tag[j];
+							}
+							li[i].getElementsByTagName('data-found')[0].style.display = 'inline';
+							dataFound.setAttribute('href', title.getAttribute('href'));
 
+							dataFound.style.display = 'inline';
+						}
+						break next;
+					}
+					else {
+						dataFound = li[i].getElementsByTagName('data-found')[0];
+						if (dataFound) {
+							dataFound.style.display = 'none';
+							title.style.font = '';
+							title.style.paddingLeft = '';
+							title.style.color = '';
+							title.innerHTML = title.innerHTML.replace("/ ", '');
+						}
+					}
+				}
+			}
+		}
+	}
+}
+
+function selectSymbols() {
+	const search = replaceSymbol(location.hash.substr(8)).toLowerCase();
```

Comment:
`replaceSymbol(location.hash.substr(8)).toLowerCase()` this is used everywhere -> copy-paste